client/trino-client/src/main/java/io/trino/client/uri/ConnectionProperties.java

```diff
@@ -704,6 +708,25 @@ public HostnameInCertificate()
         }
     }
 
+    private static class TimeZone
+            extends AbstractConnectionProperty<String, ZoneId>
+    {
+        public TimeZone()
+        {
+            super(PropertyName.TIMEZONE, NOT_REQUIRED, ALLOWED, TimeZone::parseZoneId);
+        }
+
+        private static ZoneId parseZoneId(String value)
```

Comment:
No need for this method, as `ZoneId:of` throws an unchecked exception on failure, and it gets wrapped by the property conversion code, so the user won't see the message anyway and instead sees this:

> Connection property %s value is invalid: %s

This is the original exception:

> java.time.zone.ZoneRulesException: Unknown time-zone ID: abc